controllers/auth.js

```diff
@@ -1,56 +1,111 @@
-              requestAt: new Date().toLocaleString()
-          })
-      }
-
-      //hash password
-      const saltRounds = 10
-      bcrypt.hash(password, saltRounds, function(err, hash) {
-        password = hash
-      })
-
-      const newUserData = {
-        email, 
-        password,
-        avatar
-      }
-
-      const newUserProfile = await User.create({newUserData});
-      const token = jwt.createToken(newUserProfile)
-
-      return res.status(201).json({
-        status: 201,
-        token,
-        message: "New user created, booyah!",
-        requestAt: new Date().toLocaleString()
-      });
-  
-    } catch (error) {
-      return res.status(500).json({
-        status: 500,
-        message: "Something went wrong!",
-        requestAt: new Date().toLocaleString()
+const User = require("../models/User");
+const bcrypt = require("bcrypt");
+const jwt = require("../auth/jwt");
+
+const create = async (req, res) => {
+  try {
+    let { password } = req.body;
+    const { email, avatar } = req.body;
+    const duplicateUser = await User.findOne({ email });
+
+    //test for duplicate users
+    if (duplicateUser) {
+      return res.status().json({
+        status: 400,
+        message: "User already exists!",
+        requestAt: new Date().toLocaleString(),
       });
+    }
+
+    //hash password
+    const saltRounds = 10;
+    bcrypt.hash(password, saltRounds, function (err, hash) {
+      password = hash;
+    });
+
+    const newUserData = {
+      email,
+      password,
+      avatar,
     };
-  };
+
+    const newUserProfile = await User.create({ newUserData });
+    const token = jwt.createToken(newUserProfile);
+
+    return res.status(201).json({
+      status: 201,
+      token,
+      message: "New user created, booyah!",
+      requestAt: new Date().toLocaleString(),
+    });
+  } catch (error) {
+    return res.status(500).json({
+      status: 500,
+      message: "Something went wrong!",
+      requestAt: new Date().toLocaleString(),
+    });
+  }
+};
+
+/* NOTE Login */
+const login = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+
+    //test for empty credential input
+    if (email === "" || password === "") {
+      throw "emptyForm";
+    }
+
+    const foundUser = await db.User.findOne({ email });
+
+    //test if user/email does NOT exist in the database
+    if (!foundUser) {
+      throw "invalidUser";
+    }
+
+    //test if user's password matches what's in the database
+    const isMatch = await bcrypt.compare(password, foundUser.password);
+    if (isMatch) {
+      const signedJwt = jwt.sign(
+        {
+          /* payload */ _id: foundUser._id,
+          firstName: foundUser.firstName,
+        },
+        process.env.JWT_SECRET,
+        {
+          expiresIn: "24h",
+        }
+      );
+
+      return res.status(200).json({
+        status: 200,
+        message: "Success",
+        signedJwt,
+      });
+    }
```

Comment:
For lines 70 - 86: 'jwt' in this file refers to the file auth/jwt.js; so instead of jwt.sign, you should be calling the jwt.createToken function (that is defined in the auth/jwt.js file), and pass in the parameter 'user' which in this case is the 'foundUser'.   Separately, you can update the auth/jwt.js file line 8 to add in the 'expiresIn' parameter

Reply:
Hey Rhine,
Thanks for reviewing this, & letting me know the necessary edits.
Just so I'm clear, the code needs to be updated in the following ways:
1. In 'controllers/auth.js', update lines 70-79
from**********
`      const signedJwt = jwt.sign(
        {
          /* payload */ _id: foundUser._id,
          firstName: foundUser.firstName,
        },
        process.env.JWT_SECRET,
        {
          expiresIn: "24h",
        }
      );
`
to**********
`      const signedJwt = jwt.createToken(foundUser);`

2. In 'auth/jwt.js', I'm going to add the following to line 8
from**********
`  const token = jwt.sign(payload, process.env.JWT_SECRET);`
to**********
`  const token = jwt.sign(
    {
      payload, process.env.JWT_SECRET
    },
    {
      expiresIn: "24h",
    }
);
`